include/libdnf/conf/config_main.hpp

```diff
@@ -46,32 +46,36 @@ class ConfigMain : public Config<Option::Priority::MAINCONFIG> {
     const OptionNumber<std::int32_t> & debuglevel() const;
     OptionNumber<std::int32_t> & errorlevel();
     const OptionNumber<std::int32_t> & errorlevel() const;
-    OptionString & installroot();
-    const OptionString & installroot() const;
-    OptionString & config_file_path();
-    const OptionString & config_file_path() const;
+    OptionPath & installroot();
+    const OptionPath & installroot() const;
+    OptionPath & config_file_path();
+    const OptionPath & config_file_path() const;
     OptionBool & plugins();
     const OptionBool & plugins() const;
     OptionStringList & pluginpath();
     const OptionStringList & pluginpath() const;
     OptionStringList & pluginconfpath();
     const OptionStringList & pluginconfpath() const;
-    OptionString & persistdir();
-    const OptionString & persistdir() const;
+    OptionPath & persistdir();
```

Comment:
Do we really need 3 option for storing DNF internals? And what will remain to be stored in persistdir?

Reply:
I think so. The countme data will remain in persistdir.

Reply:
Please can you make the commit header shorter? What about to remove  libdnf prefix?